src/worker.cc

```diff
@@ -1045,7 +1053,8 @@ void from_json(rapidjson::Document& doc, Options::Action action, Api& api) {
   auto matrix_locations = rapidjson::get_optional<int>(doc, "/matrix_locations");
   if (matrix_locations && (options.sources_size() == 1 || options.targets_size() == 1)) {
     options.set_matrix_locations(*matrix_locations);
-  } else if (!options.has_matrix_locations_case()) {
+  } else if (!options.has_matrix_locations_case() ||
+             (options.sources_size() != 1 && options.targets_size() != 1)) {
```

Comment:
even if the pbf has this option we still have to check if it even makes sense.

Reply:
though to be honest, I think even a many:many matrix with the 1:many timedistancematrix algorithm could want the `matrix_locations` parameter. probably one request every 10 years will want that, but it can still make sense (e.g. show me closest hotels for all these locations). so I'd actually be in favor to remove that restriction to use `matrix_locations`. 

what do you think @dnesbitt61 ?

Reply:
this could be added to more than one to many matrix requests. To me that was the most common use case (and easiest) to add. For many-to-many matrices the parameter needs further definition and documentation. Does it mean each source and target has to find that minimum number of matrix locations? I guess optimized route (traveling salesman) would not use this but if the results were used in an external optimizer that could cause issues. Anyway, I think documentation has to be explicit about how this works with many-to-many cases.

Reply:
> Does it mean each source and target has to find that minimum number of matrix locations?

I wouldn't say _has to_, but IIRC the same is true for today's 1:many, we're finished if either that number of target has been reached or all targets have been reached. in the end isn't it the exact same but just extending it to every origin (for fwd, every target for rev)? we're just using a loop there over every source or target which does the 1:many, I don't think it makes much of a difference conceptually. 

maybe there's a misunderstanding though: I'm not advocating to use this for CostMatrix, just for many:many timedistancematrix, because with CostMatrix we find bidirectional connections which makes the interpretation of such a parameter near impossible in my mind.